SteamKit2/SteamKit2/Steam/Discovery/BasicServerListProto.cs

```diff
@@ -1,13 +1,11 @@
-﻿using System.Diagnostics.CodeAnalysis;
-using ProtoBuf;
+﻿using ProtoBuf;
 
 namespace SteamKit2.Discovery
 {
     [ProtoContract]
     class BasicServerListProto
     {
         [ProtoMember(1)]
-        [DisallowNull, NotNull]
         public string? Address { get; set; }
```

Comment:
How can this be null anyway?

Reply:
good question. can you deserialize a null value or does it come back as an empty string?

Reply:
Lack of string value is deserialized as empty string. You can't distinguish between lack of string value and empty string as both are not rendered in proto. `null` value is illegal for proto strings (depending on the lib it'll fail when you attempt to assign it in proto object, or at the very latest during serialization).

Reply:
So yes, if this property is set straight from proto and never through any other means, it'll never be `null`. It can still be empty though.

Reply:
Considering only SK2 serializes this proto, it should never be empty in the first place.

Reply:
Then yeah it should be set as non-null, because you can't serialize null value in the proto, at least without resorting to custom hacks such as `var serialized = Address ?? ""`. It should just be a `string` which doesn't accept nulls, exactly the same as previous annotations (which might not be needed anymore with nullability enabled, they're redundant).